knpu/episode1/Challenges/Extending/InheritFunctionality.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Challenges\Extending;
+
+use KnpU\ActivityRunner\Activity\CodingChallenge\CodingContext;
+use KnpU\ActivityRunner\Activity\CodingChallenge\CorrectAnswer;
+use KnpU\ActivityRunner\Activity\CodingChallengeInterface;
+use KnpU\ActivityRunner\Activity\CodingChallenge\CodingExecutionResult;
+use KnpU\ActivityRunner\Activity\CodingChallenge\FileBuilder;
+
+class InheritFunctionality implements CodingChallengeInterface
+{
+    /**
+     * @return string
+     */
+    public function getQuestion()
+    {
+        return <<<EOF
+Your team has been working hard building the `DeathStar` class, only
+to find out that the Rebels have just destroyed it! Time to rebuild!
+Create a new class called `DeathStarII` and make it inherit all of
+the functionality from the original `DeathStar`. In `index.php`, instantiate
+a new `DeathStarII` object and set it to a `\$deathStar` variable.
+Long live the Empire!
+EOF;
+    }
+
+    public function getFileBuilder()
+    {
+        $fileBuilder = new FileBuilder();
+        $fileBuilder->addFileContents('index.php', <<<EOF
+<?php
+
+require 'DeathStar.php';
+require 'DeathStarII.php';
+
+// set your \$deathStar variable here
+EOF
+        );
+
+        $fileBuilder->addFileContents('DeathStar.php', <<<EOF
+<?php
+
+class DeathStar
+{
+    public function blastPlanet(\$planetName)
+    {
+        echo 'BOOM '.\$planetName;
+    }
+
+    public function getWeakness()
+    {
+        return 'Thermal Exhaust Port';
+    }
+}
+EOF
+        );
+
+        $fileBuilder->addFileContents('DeathStarII.php', <<<EOF
+
+EOF
+        );
+
+        $fileBuilder->setEntryPointFilename('index.php');
+
+        return $fileBuilder;
+    }
+
+    public function getExecutionMode()
+    {
+        return self::EXECUTION_MODE_PHP_NORMAL;
+    }
+
+    public function setupContext(CodingContext $context)
+    {
+    }
+
+    public function grade(CodingExecutionResult $result)
+    {
+        $result->assertVariableExists('deathStar');
+        $result->assertVariableEquals('deathStar', new \DeathStarII, 'The `$deathStar` variable exists, but is not set to `DeathStarII` object.');
```

Comment:
Actually, it might be simpler to do what you did below here, by accessing the variable directly to run tests on it, instead of comparing that the variable is equal to this `new \DeathstarII` object we're creating.

```php
if (!$deathstar instanceof \DeathstarII) {
    throw new GradingException('The `$deathStar` variable exists, but is not set to `DeathStarII` object.');
}
```